pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/ZKMetadataStore.java

```diff
@@ -76,6 +79,7 @@ public ZKMetadataStore(String metadataURL, MetadataStoreConfig metadataStoreConf
                         }
                     }))
                     .build();
+            zkc.addWatch("/", this::handleWatchEvent, AddWatchMode.PERSISTENT_RECURSIVE);
```

Comment:
Do you use chroot or does this really mean to watch the entire ZK tree?

Reply:
The Chroot is eventually configured on the ZK connection string. This will have 1 single watch to receive notifications for all the updates related to Pulsar/BK